packages/vaadin-grid/test/keyboard-navigation.test.js

```diff
@@ -23,69 +23,18 @@ import {
   getRowCells,
   infiniteDataProvider,
   scrollToEnd,
-  getLastVisibleItem,
-  nextResize
+  getLastVisibleItem
 } from './helpers.js';
 import '../vaadin-grid.js';
 import '../vaadin-grid-column-group.js';
 
-window.top.focus && window.top.focus();
-window.focus();
-
-if (
-  (window.chrome ||
-    /HeadlessChrome/.test(window.navigator.userAgent) ||
-    window.navigator.userAgent.toLowerCase().indexOf('firefox') > -1) &&
-  !window.document.hasFocus()
-) {
-  // Oh no! Focus kinda works, but no native events are dispatched. Let’s fake them.
-  const nativeFocus = HTMLElement.prototype.focus;
-  let fakeFocusCurrentTarget, recursiveFocusInterrupt;
-  HTMLElement.prototype.focus = function () {
-    const dispatchMockFocusEvent = function (type, bubbles, composed, target, relatedTarget) {
-      if (!target) {
-        return;
-      }
-      const e = new CustomEvent(type, { bubbles, composed: composed });
-      e.relatedTarget = relatedTarget;
-      target.dispatchEvent(e);
-    };
-
-    let activeElement,
-      recursiveFocusInvoke = false;
-    if (fakeFocusCurrentTarget) {
-      // When focus is called from a focusin/focusout listener (e. g.,
-      // recursive .focus()), activeElement might not be shifted yet.
-      // The true activeElement is a target of the previous .focus call.
-      activeElement = fakeFocusCurrentTarget;
-      recursiveFocusInvoke = true;
-    } else {
-      activeElement = document.activeElement;
-      if (activeElement) {
-        while (activeElement.shadowRoot && activeElement.shadowRoot.activeElement) {
-          activeElement = activeElement.shadowRoot.activeElement;
-        }
-      }
-    }
-
-    if (activeElement === this) {
-      // Prevent duplicate events when focus stays on the same element
-      nativeFocus.apply(this, arguments);
-    } else {
-      fakeFocusCurrentTarget = this;
-      // If shadow roots match, the events are shadow-internal, not composed
-      const composed = this.getRootNode() !== activeElement.getRootNode();
-      dispatchMockFocusEvent('focusout', true, composed, activeElement, this);
-      !recursiveFocusInterrupt && dispatchMockFocusEvent('focusin', true, composed, this, activeElement);
-      !recursiveFocusInterrupt && nativeFocus.apply(this, arguments);
-      !recursiveFocusInterrupt && dispatchMockFocusEvent('blur', false, composed, activeElement, this);
-      !recursiveFocusInterrupt && dispatchMockFocusEvent('focus', false, composed, this, activeElement);
-      fakeFocusCurrentTarget = undefined;
-    }
-
-    // Recursive focus invoke should interrupt the parent
-    recursiveFocusInterrupt = recursiveFocusInvoke;
-  };
+async function isTestRunnerControlledBrowser() {
```

Comment:
Do we really need this? We use `sendKeys` in many places, and debugging is still possible when using built-in debugger (available both for Chrome and Playwright), instead of using manual mode.

Reply:
True, removed the workaround.